Gemfile

```
@@ -1,6 +1,9 @@
 # foreman plugins import this file therefore __FILE__ cannot be used
 FOREMAN_GEMFILE = __FILE__ unless defined? FOREMAN_GEMFILE
 
+# pin YAML library before we even load YAML configuration
+gem 'psych', '~> 3.3'
```

I'd prefer to be explicit that we don't support version 4 but continue to accept all older versions. This is more flexible on the packaging side
```suggestion
gem 'psych', '< 4'
```